core/src/main/java/org/mule/runtime/core/internal/util/rx/RxUtils.java

```diff
@@ -406,4 +483,74 @@ public static <T> FluxSinkSupplier<T> createRoundRobinFluxSupplier(Function<Flux
     }, identity());
   }
 
+  public static class MultiFluxSubscriber<T> implements CoreSubscriber<T> {
+
+    public static String MULTI_FLUX_SUBSCRIBER = "MULTI_FLUX_SUBSCRIBER";
+
+    private Throwable lastSubscriptionError;
+
+    private final Latch completionLatch;
+    private final Phaser subscriptionPhaser = new Phaser(1);
+    private boolean subscribed = false;
+
+    public MultiFluxSubscriber(Latch completionLatch) {
+      this.completionLatch = completionLatch;
+    }
+
+    @Override
+    public Context currentContext() {
+      return CoreSubscriber.super.currentContext().put(MULTI_FLUX_SUBSCRIBER, this);
+    }
+
+    @Override
+    public void onSubscribe(Subscription s) {
+      // Protect the subscriptionPhaser from potential extra calls.
+      synchronized (this) {
+        if (!subscribed) {
```

Comment:
Can we add a logger with a warning a case extra calls happen, because a far as I understand it would be an unexpected behavior.

Reply:
possible internal reactor behavior as discussed.